src/jit/morph.cpp

```diff
@@ -204,6 +204,7 @@ GenTreePtr Compiler::fgMorphCast(GenTreePtr tree)
                 {
                     case TYP_INT:
 #ifdef _TARGET_X86_ // there is no rounding convert to integer instruction on ARM or x64 so skip this
+#ifdef LEGACY_BACKEND
```

Comment:
This could use a brief comment along the lines of "the RyuJIT backend does not use the x87 FPU and therefore does not support folding the cast with the `CORINFO_INTRINSIC_Round` intrinsic".